optimizer/lib/transformers/AddBlurryImagePlaceholders.js

```diff
@@ -34,24 +40,37 @@ class AddBlurryImagePlaceholders {
     const html = tree.root.firstChildByTag('html');
     const body = html.firstChildByTag('body');
     const promises = [];
+    let currPlaceholderCount = 0;
     for (let node = body; node !== null; node = node.nextNode()) {
       const {tagName} = node;
       let src;
+      if (tagName === 'template') {
+        node = nextNode(node);
+        continue;
+      }
       if (tagName === 'amp-img') {
         src = node.attribs.src;
       }
       if (tagName === 'amp-video' && node.attribs.poster) {
         src = node.attribs.poster;
       }
-      if (src && !this.hasPlaceholder_(node)) {
-        promises.push(this.addBlurryPlaceholder_(tree, src).then((imgChild) => {
-          node.appendChild(imgChild);
-        }));
-      }
+      if (currPlaceholderCount < MAX_BLURRED_PLACEHOLDERS && src &&
+        !this.hasPlaceholder_(node) &&
```

Comment:
Can you move the checks for `src && !this.hasPlaceholder_(node)` into `this.shouldAddBlurryPlaceholder_`